packages/fx-core/src/plugins/resource/bot/v2/mapping.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ServiceType } from "../../../../common/azure-hosting/interfaces";
+import { BotNotificationTriggers } from "../../../solution";
+import { TemplateProjectsScenarios } from "../constants";
+import { HostTypes } from "../resources/strings";
+import { ProgrammingLanguage } from "./enum";
+
+const runtimeMap: Map<string, string> = new Map<string, string>([
+  [ProgrammingLanguage.Js, "node"],
+  [ProgrammingLanguage.Ts, "node"],
+  [ProgrammingLanguage.Csharp, "csharp"],
+]);
+
+const serviceMap: Map<string, ServiceType> = new Map<string, ServiceType>([
+  [HostTypes.APP_SERVICE, ServiceType.AppService],
+  [HostTypes.AZURE_FUNCTIONS, ServiceType.Functions],
+]);
+
+const langMap: Map<string, string> = new Map<string, string>([
+  ["javascript", ProgrammingLanguage.Js],
+  ["typescript", ProgrammingLanguage.Ts],
+  ["csharp", ProgrammingLanguage.Csharp],
+]);
+
+const triggerScenarioMap: Map<string, string> = new Map<string, string>([
```

Comment:
This could be as below to include all notification scnearios
```
const triggerScenarioMap: Map<string, string[]> = new Map<string, string[]>([
[
    BotNotificationTriggers.Http,
    [TemplateProjectsScenarios.NOTIFICATION_FUNCTION_TRIGGER_HTTP_SCENARIO_NAME, TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME],
  ],
  [
    BotNotificationTriggers.Timer,
    [TemplateProjectsScenarios.NOTIFICATION_FUNCTION_TRIGGER_TIMER_SCENARIO_NAME, TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME],
  ],
  [ 
     AppServiceOptionItem.id
    [TemplateProjectsScenarios.NOTIFICATION_RESTIFY_SCENARIO_NAME],
  ],
```